backend/api/main/views.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+from django.http.response import HttpResponse, JsonResponse
+from rest_framework.views import APIView
+from rest_framework.parsers import FormParser
+import json
+from api.main.tree import tree_to_json
+from core.models import Course
+
+
+def main_data(request):
+    if request.method != 'GET':
+        return HttpResponse(status=405)  # Method Not Allowed
+    tree = request.session.get('tree', None)
+    if tree is None:
+        return HttpResponse(status=400)  # Bad Request
+    return JsonResponse(tree_to_json(tree))
+
+
+class SearchModal(APIView):
+    parser_classes = (FormParser,)
+
+    def post(self, request):
+        tree = request.session.get('tree', None)
+        node_title = request.session.get('node_title', None)
+        if tree is None or node_title is None:
+            return HttpResponse(status=400)  # Bad Request
```

Comment:
Check [this](https://docs.djangoproject.com/en/1.10/ref/request-response/#django.http.HttpResponseBadRequest) out